src/actions/index.ts

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { assign, AssignAction, raise, sendTo } from 'xstate';
+import { Context, Event } from '../machines/types';
+import { get } from 'lodash';
+import logger from '../logger';
+import { hashTxData } from '../utils';
+
+export const storeInitialState = assign({
+  initialEventId: (_context: Context, event: Event) => {
+    // @ts-ignore
+    return event.data.lastEventId;
+  },
+});
+
+export const unwrapEvent = assign({
+  event: (_context: Context, event: Event) => {
+    if (event.type !== 'METADATA_DECIDED') {
+      return;
```

Comment:
I guess `context.event` would work here, to return the old event, but I think we should throw because this should never happen

Done in 420973d25f648f4995ac75a8a74789445a4ceda3